### PR TITLE
Add header to link examples to docs

### LINKED ISSUE #133: Explain example at the top of the example file
Following this [comment](https://github.com/mapbox/mapbox-navigation-ios-examples/pull/106#discussion_r674702582): the examples contain code that is not relevant to understanding the example code's behavior. We should explain the example at the beginning of the example file so developers spend less time looking through example code to find the relevant code for their desired behavior.

/cc: @S2Ler @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by pjleonard37**
Closes #133 

This PR adds a brief header to each example (see below) that links to the Mapbox documentation page. This header will appear in the example code in the repo and in the code snippet on the docs examples page. Additional changes will add descriptions to each docs examples page. 

>  This code example is part of the Mapbox Navigation SDK for iOS demo app,
>  which you can build and run: https://github.com/mapbox/mapbox-navigation-ios-examples
>  To learn more about each example in this app, including descriptions and links
>  to documentation, see our docs: https://docs.mapbox.com/ios/navigation/examples

### CODE REVIEW

File: Navigation-Examples/Examples/Advanced.swift [hunk 1]
```diff
@@ -1,3 +1,10 @@
+/*
+ This code example is part of the Mapbox Navigation SDK for iOS demo app,
+ which you can build and run: https://github.com/mapbox/mapbox-navigation-ios-examples
+ To learn more about each example in this app, including descriptions and links
+ to documentation, see our docs: https://docs.mapbox.com/ios/navigation/examples
```
Comment (1ec5) [inline Navigation-Examples/Examples/Advanced.swift:5]: The Swift files aren’t always named very consistently with the URL slugs or example names on the published site. To help users find what they’re looking for, let’s link directly to the corresponding published example.

**Review by 1ec5 — COMMENTED** [review]
Thanks for taking care of this task!

File: Navigation-Examples/Examples/Advanced.swift (hunk 1 above)
Comment (pjleonard37) [reply Navigation-Examples/Examples/Advanced.swift]: 👍  - that seems like a good idea. I updated links for those examples which have a specific published example (so all except Beta Query Parameters and Offline Regions). Once those are added to the Docs site I can add in the new links.

**Review by 1ec5 — APPROVED** [review]
Thanks!

**Pull request merged**